src/transaction/mvcc_table.cpp

```diff
@@ -298,14 +298,18 @@ mvcctable::build_mvcc_info (log_tdes &tdes)
       trans_status.m_active_mvccs.copy_to (tdes.mvccinfo.snapshot.m_active_mvccs,
 					   mvcc_active_tran::copy_safety::THREAD_UNSAFE);
       /* load statistics temporary disabled need to be enabled when activate count optimization */
-#if 0
       /* load global statistics. This must take place here and nowhere else. */
-      if (logtb_load_global_statistics_to_tran (thread_p) != NO_ERROR)
+      tdes.mvccinfo.snapshot.valid = true;
+      /*
+       snapshot should be set to valid to avoid the endless mvcc_info build
+       because gtb_load_global_statistics_to_tran calls build_mvcc_info recursively
+       since the valid flag is false.
+      */
+      if (logtb_load_global_statistics_to_tran (thread_get_thread_entry_info())!= NO_ERROR)
 	{
 	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_MVCC_CANT_GET_SNAPSHOT, 0);
-	  error_code = ER_MVCC_CANT_GET_SNAPSHOT;
+	  return;
```

Comment:
Is it ok to return here? Callers don't check the error code of `build_mvcc_info()` and assums the mvcc info is built. It seems that it will end up unitialized `tdes.mvccinfo`.

Reply:
It seems not to be valid code. I will fix to reset the snapshot.valid flag to false.

Reply:
It seems "return" is not vaild here. It's better to leave the errorce without returning immediately.